### ISSUE TITLE
Inferencing example can not read/write to HDFS

### ISSUE THREAD

**Issue by earthquakesan** [SANSA-Stack/SANSA-Examples#10]
Running this code sample results in failure:
```
import java.io.File
import scala.collection.mutable
import org.apache.spark.sql.SparkSession
import net.sansa_stack.rdf.spark.model.JenaSparkRDDOps
import net.sansa_stack.inference.spark.RDFGraphMaterializer
import net.sansa_stack.inference.spark.data.RDFGraphLoader
import net.sansa_stack.inference.spark.forwardchaining.ForwardRuleReasonerRDFS
import net.sansa_stack.inference.rules.ReasoningProfile
import net.sansa_stack.inference.spark.forwardchaining.ForwardRuleReasonerOWLHorst
import net.sansa_stack.inference.rules.ReasoningProfile._
import net.sansa_stack.inference.spark.data.RDFGraphWriter

// load triples from disk
val input = "hdfs://namenode:8020/data/rdf.nt"
val output = "hdfs://namenode:8020/data/output/"
val argprofile = "rdfs"
val profile = argprofile match {
  case "rdfs"      => ReasoningProfile.RDFS
  case "owl-horst" => ReasoningProfile.OWL_HORST
}

val graph = RDFGraphLoader.loadFromFile(new File(input).getAbsolutePath, sc, 4)
println(s"|G|=${graph.size()}")

// create reasoner
val reasoner = profile match {
  case RDFS      => new ForwardRuleReasonerRDFS(sc)
  case OWL_HORST => new ForwardRuleReasonerOWLHorst(sc)
}

// compute inferred graph
val inferredGraph = reasoner.apply(graph)
println(s"|G_inferred|=${inferredGraph.size()}")

// write triples to disk
RDFGraphWriter.writeGraphToFile(inferredGraph, new File(output).getAbsolutePath)
```
Here is the stacktrace:
```
import java.io.File
import scala.collection.mutable
import org.apache.spark.sql.SparkSession
import net.sansa_stack.rdf.spark.model.JenaSparkRDDOps
import net.sansa_stack.inference.spark.RDFGraphMaterializer
import net.sansa_stack.inference.spark.data.RDFGraphLoader
import net.sansa_stack.inference.spark.forwardchaining.ForwardRuleReasonerRDFS
import net.sansa_stack.inference.rules.ReasoningProfile
import net.sansa_stack.inference.spark.forwardchaining.ForwardRuleReasonerOWLHorst
import net.sansa_stack.inference.rules.ReasoningProfile._
import net.sansa_stack.inference.spark.data.RDFGraphWriter
input: String = hdfs://namenode:8020/data/rdf.nt
output: String = hdfs://namenode:8020/data/output/
argprofile: String = rdfs
profile: net.sansa_stack.inference.rules.ReasoningProfile.Value = RDFS
graph: net.sansa_stack.inference.spark.data.RDFGraph = RDFGraph(MapPartitionsRDD[25] at map at RDFGraphLoader.scala:28)
java.lang.IllegalArgumentException: java.net.URISyntaxException: Expected scheme-specific part at index 5: hdfs:
  at org.apache.hadoop.fs.Path.initialize(Path.java:205)
  at org.apache.hadoop.fs.Path.<init>(Path.java:171)
  at org.apache.hadoop.fs.Path.<init>(Path.java:93)
  at org.apache.hadoop.fs.Globber.glob(Globber.java:211)
  at org.apache.hadoop.fs.FileSystem.globStatus(FileSystem.java:1676)
  at org.apache.hadoop.mapred.FileInputFormat.singleThreadedListStatus(FileInputFormat.java:259)
  at org.apache.hadoop.mapred.FileInputFormat.listStatus(FileInputFormat.java:229)
  at org.apache.hadoop.mapred.FileInputFormat.getSplits(FileInputFormat.java:315)
  at org.apache.spark.rdd.HadoopRDD.getPartitions(HadoopRDD.scala:202)
  at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:252)
  at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:250)
  at scala.Option.getOrElse(Option.scala:121)
  at org.apache.spark.rdd.RDD.partitions(RDD.scala:250)
  at org.apache.spark.rdd.MapPartitionsRDD.getPartitions(MapPartitionsRDD.scala:35)
  at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:252)
  at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:250)
  at scala.Option.getOrElse(Option.scala:121)
  at org.apache.spark.rdd.RDD.partitions(RDD.scala:250)
  at org.apache.spark.rdd.MapPartitionsRDD.getPartitions(MapPartitionsRDD.scala:35)
  at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:252)
  at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:250)
  at scala.Option.getOrElse(Option.scala:121)
  at org.apache.spark.rdd.RDD.partitions(RDD.scala:250)
  at org.apache.spark.rdd.MapPartitionsRDD.getPartitions(MapPartitionsRDD.scala:35)
  at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:252)
  at org.apache.spark.rdd.RDD$$anonfun$partitions$2.apply(RDD.scala:250)
  at scala.Option.getOrElse(Option.scala:121)
  at org.apache.spark.rdd.RDD.partitions(RDD.scala:250)
  at org.apache.spark.SparkContext.runJob(SparkContext.scala:1965)
  at org.apache.spark.rdd.RDD.count(RDD.scala:1158)
  at net.sansa_stack.inference.spark.data.RDFGraph.size(RDFGraph.scala:68)
  ... 52 elided
Caused by: java.net.URISyntaxException: Expected scheme-specific part at index 5: hdfs:
  at java.net.URI$Parser.fail(URI.java:2848)
  at java.net.URI$Parser.failExpecting(URI.java:2854)
  at java.net.URI$Parser.parse(URI.java:3057)
  at java.net.URI.<init>(URI.java:746)
  at org.apache.hadoop.fs.Path.initialize(Path.java:202)
  ... 82 more
```

**Comment by earthquakesan** [comment]
Still does not work:
```
inferencing_1                    | Caused by: java.lang.ArrayIndexOutOfBoundsException: 2
inferencing_1                    | 	at net.sansa_stack.inference.utils.NTriplesStringToRDFTriple.apply(NTriplesStringToRDFTriple.scala:15)
inferencing_1                    | 	at net.sansa_stack.inference.utils.NTriplesStringToRDFTriple.apply(NTriplesStringToRDFTriple.scala:10)
inferencing_1                    | 	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
inferencing_1                    | 	at org.apache.spark.util.Utils$.getIteratorSize(Utils.scala:1762)
inferencing_1                    | 	at org.apache.spark.rdd.RDD$$anonfun$count$1.apply(RDD.scala:1158)
inferencing_1                    | 	at org.apache.spark.rdd.RDD$$anonfun$count$1.apply(RDD.scala:1158)
inferencing_1                    | 	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:1951)
inferencing_1                    | 	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:1951)
inferencing_1                    | 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
inferencing_1                    | 	at org.apache.spark.scheduler.Task.run(Task.scala:99)
inferencing_1                    | 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:322)
inferencing_1                    | 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
inferencing_1                    | 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
inferencing_1                    | 	at java.lang.Thread.run(Thread.java:745)
```

**Comment by LorenzBuehmann** [comment]
Both issues have been solved in 0.1.1-SNAPSHOT version of SANSA-Inference.

Fixed by commit https://github.com/SANSA-Stack/SANSA-Examples/commit/6dd895694fc7c942f2d9fafe3233e4cfd2edc8c2
